priml-examples/cv-prod-cons-err.prm

```diff
@@ -0,0 +1,93 @@
+(* * * * * * * * * * * * * * * * * * * * * * * * * * * * * *
+ * producer consumer priority inversion error example
+ * ---
+ *
+ * description quoted & code translated from the C-family example (Fig. 3)
+ * given in _pldi23_:
+ *
+ * > Initial attempt at the producer-consumer program, which has a priority
+ * > inversion (and a type error, in our full system)
+ *
+ * ```
+ * void <High>prod(CV<High> cv, Buffer *buf){
+ *   while(true) {
+ *     ...
+ *     append(buf, x);
+ *     signal(cv);
+ *   }
+ * }
+ * 
+ * void <High>cons(CV<High> cv, Buffer *buf){
+ *   while(true) {
+ *     if empty(buffer){
+ *       wait(cv);
+ *     }
+ *     x = pop(buf);
+ *     ...
+ *   }
+ * }
+ * 
+ * void <Low>main(){
+ *   CV<High>cv = new CV();
+ *   Buffer *buf;
+ *   
+ *   spawn<High>(cons(cv, buf));
+ *   spawn<High>(prod(cv, buf)); // ill-typed
+ * }
+ * ```
+ * * * * * * * * * * * * * * * * * * * * * * * * * * * * * *)
+
+priority high
+priority low
+order low < high
+
+(* prod
+ * ---
+ *
+ * increments a number and pushes it to a queue, then signals before looping
+ *)
+(* FIXME: how to specify queue as a reference to a list of ints *)
+fun[high] prod (cv: cv[high]) (queue: ref int list) =
```

Comment:
Not sure I'm specifying the type correctly here for `queue`. I'm trying to use a reference to a mutable list of integers as a simple work queue that producer adds values to for consumer to use.

Reply:
```suggestion
fun[high] prod (cv: cv[high]) (queue: int list ref) =
```